kedro/extras/extensions/ipython.py

```diff
@@ -77,35 +89,21 @@ def reload_kedro(path, env: str = None, extra_params: Dict[str, Any] = None):
         logging.info("Registered line magic `%s`", line_magic.__name__)  # type: ignore
 
 
-def init_kedro(path=""):
-    """Line magic to set path to Kedro project.
-    `%reload_kedro` will default to this location.
-    """
-    global project_path
-    if path:
-        project_path = Path(path).expanduser().resolve()
-        logging.info("Updated path to Kedro project: %s", str(project_path))
-    else:
-        logging.info("No path argument was provided. Using: %s", str(project_path))
-
-
 def load_ipython_extension(ipython):
     """Main entry point when %load_ext is executed"""
 
-    global project_path
-    global startup_path  # pylint:disable=global-variable-not-assigned
+    global default_project_path
 
-    ipython.register_magic_function(init_kedro, "line")
     ipython.register_magic_function(reload_kedro, "line", "reload_kedro")
 
-    project_path = _find_kedro_project(startup_path)
+    default_project_path = _find_kedro_project(Path.cwd())
 
     try:
-        reload_kedro(project_path)
+        reload_kedro(default_project_path)
     except (ImportError, ModuleNotFoundError):
         logging.error("Kedro appears not to be installed in your current environment.")
     except Exception:  # pylint: disable=broad-except
         logging.warning(
-            "Kedro extension was registered. Make sure you pass the project path to "
-            "`%reload_kedro` or set it using `%init_kedro`."
+            "Kedro extension was registered but couldn't find a Kedro project. "
+            "Make sure you run `%reload_kedro <path_to_kedro_project>`."
         )
```

Comment:
Minor: maybe this made sense before but it feels like a weird way of handling the flow now. If I try loading the extension and don't have kedro installed then I'll get "ModuleNotFoundError: No module named 'kedro.extras'" which is pretty clear already so I don't see the need to catch the `ImportError` at all here? And then putting everything else in a broad except and assuming that's because a kedro project can't be found seems odd given we have a special value `None` that signifies exactly that already.

To me this would make more sense (combined it with my above suggestion and tidied the result up a bit) as something like this:
```
kedro_project_path = _find_kedro_project(Path.cwd())
if kedro_project_path is not None:
    reload_kedro(kedro_project_path)
else:
    logging.warning(
            "Kedro extension was registered but couldn't find a Kedro project. "
            "Make sure you run `%reload_kedro <path_to_kedro_project>`."
        )
```

Reply:
If I read this correctly, then it's not actually true. `None` is returned if it finds a kedro project, not if kedro is installed. The import and module errors can still be raised if there IS a kedro project, but kedro is not installed (wrong activated environment for instance)  in which  case we give a more friendly errr message to the user.

Reply:
Basically if someone has the kedro ipython extension loaded (e.g.set in the ipython config), it *shouldn't* prevent them from using ipython at all. That's why we catch the exception and move on, but give them a heads up.

Reply:
In the spirit of [this principle](https://mtlynch.io/human-code-reviews-2/#aim-to-bring-the-code-up-a-letter-grade-or-two), merging this as is as it's already gone a little out of scope, and final perfecting/refactoring can be done under a different task, as that will also require updating unit tests..